debezium-embedded/src/main/java/io/debezium/embedded/ConverterBuilder.java

```diff
@@ -148,8 +126,48 @@ private boolean shouldConvertHeadersToString() {
         return isFormat(formatHeader, Json.class);
     }
 
-    private List<Header<byte[]>> convertHeaders(
-                                                SourceRecord record, String topicName, HeaderConverter headerConverter) {
+    private boolean shouldPassHeaders() {
+        return config.subset(CONVERTER_PREFIX, true).getBoolean(PASS_HEADERS, false);
+    }
+
+    private KeyValueHeaders convertRecord(SourceRecord record, HeaderConverter headerConverter) {
+        Converter keyConverter = createConverter(formatKey, true);
+        Converter valueConverter = createConverter(formatValue, false);
+        String topicName = Objects.requireNonNullElse(record.topic(), TOPIC_NAME);
+
+        // If your key/value converters don't require headers for read or write, we can skip the header conversion to
+        // save some CPU cycles
+        if (shouldPassHeaders()) {
```

Comment:
I believe this is unnecessary optimization. If the header converter is present then the conversion and passing of headers should be done always.
It will also simplify the code

Reply:
Hi Jiri, Thanks for the review.

Sounds perfect to me - thoughts on this kind of change instead?
https://github.com/debezium/debezium/pull/5702#issuecomment-2246719521

Reply:
👋 There is a different implementation on this branch: https://github.com/debezium/debezium/compare/main...ryanvanhuuksloot:debezium:rvh.example-record-headers

It was the first pass at implementing but I was asked to do this without always passing headers.
But it always passes the headers and I believe is more aligned with what you are asking. If you look and think it is better I can swap this PR to use something more akin to that.

Reply:
@jpechane did that make more sense?